shotover-proxy/src/message/mod.rs

```diff
@@ -377,26 +400,39 @@ impl Value {
             match spec.col_type.id {
                 ColType::Ascii => Value::Strings(decode_ascii(actual_bytes).unwrap()),
                 ColType::Bigint => Value::Integer(decode_bigint(actual_bytes).unwrap()),
-                ColType::Blob => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
                 ColType::Boolean => Value::Boolean(decode_boolean(actual_bytes).unwrap()),
-                ColType::Counter => Value::Integer(decode_int(actual_bytes).unwrap() as i64),
-                ColType::Decimal => unimplemented!("We dont have a decimal type yet"),
+                ColType::Counter | ColType::Int => {
+                    Value::Integer(decode_int(actual_bytes).unwrap() as i64)
+                }
+                ColType::Decimal => {
+                    let decimal = decode_decimal(actual_bytes).unwrap();
+                    let big_decimal = BigDecimal::new(decimal.unscaled, decimal.scale.into());
+                    Value::Decimal(big_decimal)
+                }
                 ColType::Double => Value::Float(decode_double(actual_bytes).unwrap()),
                 ColType::Float => Value::Float(decode_float(actual_bytes).unwrap() as f64),
-                ColType::Int => Value::Integer(decode_int(actual_bytes).unwrap() as i64),
-                ColType::Uuid => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
                 ColType::Varchar => Value::Strings(decode_varchar(actual_bytes).unwrap()),
-                ColType::Varint => unimplemented!("We dont have a varint type yet"),
-                ColType::Timeuuid => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
+                ColType::Varint => Value::Varint(decode_varint(actual_bytes).unwrap()),
+                ColType::Timeuuid => Value::Timeuuid(decode_timeuuid(actual_bytes).unwrap()),
+                ColType::Blob => Value::Bytes(Bytes::copy_from_slice(actual_bytes)),
+                ColType::Uuid => Value::Uuid(uuid::Uuid::from_slice(actual_bytes).unwrap()),
                 ColType::Inet => Value::Inet(decode_inet(actual_bytes).unwrap()),
-                ColType::Timestamp => Value::NULL,
-                ColType::Date => Value::NULL,
-                ColType::Time => Value::NULL,
+                ColType::Date => Value::Date(decode_date(actual_bytes).unwrap()),
+                ColType::Timestamp => Value::Timestamp(decode_timestamp(actual_bytes).unwrap()),
+                ColType::Time => Value::Time(decode_time(actual_bytes).unwrap()),
                 ColType::Smallint => Value::Integer(decode_smallint(actual_bytes).unwrap() as i64),
                 ColType::Tinyint => Value::Integer(decode_tinyint(actual_bytes).unwrap() as i64),
-                // TODO: process collection types based on ColTypeOption
-                // (https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L569)
-                _ => Value::NULL,
+                ColType::Custom => Value::Strings(decode_custom(actual_bytes).unwrap()),
```

Comment:
Collides with ColType::Varchar

Reply:
What collides?

Reply:
They both converted to Value::Strings which would make it impossible to unambiguously return trip.
Problem is resolved now though because ColType::Custom now is now `unimplemented!()`